W3ChampionsStatisticService/WebApi/ActionFilters/CheckIfBattleTagBelongsToAuthCodeFilter.cs

```diff
@@ -22,21 +24,25 @@ public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionE
             if (queryString.AllKeys.Contains("authorization"))
             {
                 var auth = queryString["authorization"];
-                var res = _authService.GetUserByToken(auth);
+                try {
+                    var res = _authService.GetUserByToken(auth, false);
+                    var btagString = battleTag?.ToString();
 
-                var btagString = battleTag?.ToString();
-                if (
-                    res != null
-                    && !string.IsNullOrEmpty(btagString)
-                    && btagString.Equals(res.BattleTag))
-                {
-                    context.ActionArguments["battleTag"] = res.BattleTag;
-                    await next.Invoke();
+                    if (!string.IsNullOrEmpty(btagString) && btagString.Equals(res.BattleTag))
+                    {
+                        context.ActionArguments["battleTag"] = res.BattleTag;
+                        await next.Invoke();
+                    }
+                }
+                catch (SecurityTokenExpiredException) {
+                    var unauthorizedResult = new UnauthorizedObjectResult(new ErrorResult("Token expired."));
+                    context.Result = unauthorizedResult;
+                }
+                catch (Exception) {
```

Comment:
Need to handle an exception if res is null, it will return 401 and it won't be clear what happened.
If Res = null, we should return 502, which will indicate the issue is in ID service.

Reply:
res won't be null anymore. it will either be a `W3CUserAuthenticationDto` or some exception has been thrown.